chain/messagepool/messagepool.go

```diff
@@ -474,8 +475,8 @@ func (mp *MessagePool) TryForEachPendingMessage(f func(cid.Cid) error) error {
 
 func (mp *MessagePool) resolveToKey(ctx context.Context, addr address.Address) (address.Address, error) {
 	// check the cache
```

Comment:
Can we just check whether the sender is a robust address (anything except an ID address is robust) here, and return if so? That will:

- be faster
- reduce the size of this cache by half, because we can drop `mp.keyCache.Add(ka, ka)` on line 491.

Reply:
(seems like something that can be done in a future patch)

Reply:
Thanks - I have logged this as the second item in this enhancement issue: https://github.com/filecoin-project/lotus/issues/10597